beacon-chain/rpc/service.go

```diff
@@ -259,22 +259,28 @@ func (s *Service) ValidatorShardID(ctx context.Context, req *pb.PublicKey) (*pb.
 	return &pb.ShardIDResponse{ShardId: shardID}, nil
 }
 
-// ValidatorSlot is called by a validator to get the slot number of when it's suppose
-// to proposer or attest.
-func (s *Service) ValidatorSlot(ctx context.Context, req *pb.PublicKey) (*pb.SlotResponse, error) {
+// ValidatorSlotAndResponsibility fetches a validator's assigned slot number
+// and whether it should act as a proposer/attester.
+func (s *Service) ValidatorSlotAndResponsibility(ctx context.Context, req *pb.PublicKey) (*pb.SlotResponsibilityResponse, error) {
 	cState := s.chainService.CurrentCrystallizedState()
 
-	slot, err := casper.ValidatorSlot(
+	slot, responsibility, err := casper.ValidatorSlotAndResponsibility(
 		req.PublicKey,
 		cState.CurrentDynasty(),
 		cState.Validators(),
 		cState.ShardAndCommitteesForSlots(),
 	)
+	var role pb.ValidatorRole
+	if responsibility == "proposer" {
+		role = pb.ValidatorRole_PROPOSER
+	} else if responsibility == "attester" {
+		role = pb.ValidatorRole_ATTESTER
+	}
 	if err != nil {
```

Comment:
shouldn't this err check be right after `casper.ValidatorSlotAndResponsibility`. Line 273?

Reply:
good point